Tests/Tests/STPRedirectContextTest.m

```diff
@@ -224,47 +224,6 @@ - (void)testSafariViewControllerRedirectFlow_foregroundNotification {
     [self unsubscribeContext:context];
 }
 
-
-/**
- After starting a SafariViewController redirect flow,
- when the shared URLCallbackHandler is called with a valid URL,
- RedirectContext's completion block and dismiss method should be called.
- */
-- (void)testSafariViewControllerRedirectFlow_callbackHandlerCalledValidURL {
-    id mockVC = OCMClassMock([UIViewController class]);
-    STPSource *source = [STPFixtures iDEALSource];
-    XCTestExpectation *exp = [self expectationWithDescription:@"completion"];
-    STPRedirectContext *context = [[STPRedirectContext alloc] initWithSource:source completion:^(NSString *sourceID, NSString *clientSecret, NSError *error) {
-        XCTAssertEqualObjects(sourceID, source.stripeID);
-        XCTAssertEqualObjects(clientSecret, source.clientSecret);
-        XCTAssertNil(error);
-        [exp fulfill];
-    }];
-    XCTAssertEqualObjects(source.redirect.returnURL, context.returnURL);
-    id sut = OCMPartialMock(context);
-
```

Comment:
I'm guessing the issue with these tests is that the completion block isn't being called, because now that will only happen after `safariViewControllerDidCompleteDismissal:` is called?  

To keep these tests, could we try adding something like:

```
OCMStub([sut handleRedirectCompletionWithError:[OCMArg any] shouldDismissViewController:YES]).andForwardToRealObject().andDo(^(__unused NSInvocation *invocation) {
  [sut safariViewControllerDidCompleteDismissal:nil];
}
```

Reply:
I fixed all the tests using your suggested approach.